(ADDED) Squidink-Frontend Components

RichardMcCarty1 opened the pull request.

bowring commented:
This compiles and runs.  However, in browser I get the right hand side but the left hand side (filebrowser) tries to connect and then says '192.168.86.29 took too long to respond.'   I need instructions for running file browser?

bowring commented:
Even with rice embed-go   running in http, I get the same error.

RichardMcCarty1 commented:
That's because your filebrowser endpoint is different from what the i-frame expects, If you go to src/constants/api there should be a constant called FILEBROWSER_URL, if you change that to 127.0.0.1:8080 it should pop up.

Just as a fore-warning, it will probably only pop up the log-in menu, not the full folder view. If you'd like to see what its like with the  folder view you can log into Filebrowser in a separate tab then refresh Squidink and it should display.

bowring commented:
still get same error although it first says trying to connect to localhost;

RichardMcCarty1 commented:
Is it still including the http preceding the IP such as "http://127.0.0.1:8080"?

Also is the filebrowser executable running such that its publishing on http://127.0.0.1:8080?

bowring commented:
I made the constant "127.0.0.1:8080". and the connecting message says "localhost"

not sure what second question refers to as I just used rice ... am I supposed to do something else

RichardMcCarty1 commented:
You'll need to add the characters "http://" preceding the constant IP in the API folder. Javascript / HTML by their most default nature don't automatically apply the prefix if I remember correctly.

For the filebrowser, rice simply serves as a way for GO to package all of the static frontend components neatly, so images and .js files and the sort would be packaged for GO to then publish accordingly. 

To actually run the filebrowser as a page such that you can access it, you have to go to the the parent directory (The one preceeding frontend or http or any of the other primary folders) and run filebrowser. For me, on my wiindows machine, it is called filebrowser.exe so simply inputting that opens it up. I'm pretty sure for you, you did "open filebrowser" in our last meeting.

bowring commented:
lost the chat notes from google call;  anyway browser opens in port 8080 but still get error on left hand panel of squidink

RichardMcCarty1 commented:
The most recent push should fix your issues. The I-Frame wasn't referencing the constant file, must've forgotten, my apologies.

So long as your Filebrowser is running you should be able to open up Squidink and see the log-in page in the far left. Ultimately it will resolve to the file structure once we route Squidink clicks to the filebrowser and then back to the Squidink endpoint, but in its current state you will just see the login page.

bowring commented:
I am getting an infinite loop on the right switching between SquidInk and "your file is loading". and a flashing dot on the left

bowring commented:
@RichardMcCarty1  - what's up with the package.json conflict?

RichardMcCarty1 commented:
@bowring Just resolved it, was a slight mismatch because I removed some unused dependencies.

bowring commented:
➜  CirdlesWebUI git:(RichardMcCarty1-packageConflictFix) ✗ less /Users/bowring/.npm/eresolve-report.txt

      }
    ]
  },
  "currentEdge": {
    "type": "prod",
    "name": "react",
    "spec": "^17.0.1",
    "from": {
      "location": "/Users/bowring/Development/CirdlesWebUI"
    }
  },
  "edge": {
    "type": "peer",
    "name": "react",
    "spec": "^16.14.0",
    "error": "INVALID",
    "from": {
      "name": "react-dom",
      "version": "16.14.0",
      "whileInstalling": {
        "name": "cirdlesweb",
        "version": "0.1.0",
        "path": "/Users/bowring/Development/CirdlesWebUI"
      },
      "location": "node_modules/react-dom",
      "dependents": [
        {
          "type": "prod",
          "name": "react-dom",
          "spec": "^16.13.1",
          "from": {
            "location": "/Users/bowring/Development/CirdlesWebUI"
          }
        }
      ]
    }
  },
  "peerConflict": null,
  "strictPeerDeps": false,
  "force": false
}

bowring commented:
did some forcing and got it to take

bowring commented:
seeing this:
<img width="1581" alt="squidink" src="https://user-images.githubusercontent.com/1092963/119031141-bf33f700-b978-11eb-925b-2f382afe4773.png">

RichardMcCarty1 commented:
From the looks of it, the issue is caused by dependency versioning, see [here](https://stackoverflow.com/questions/43249820/npm-install-gives-error-message-invalid-what-is-it). Your forcing then seems to remove most if not all of the styling associated with the components. Did you happen to run an update on the dependencies?

bowring commented:
i will try another clean clone

bowring commented:
same problem ... do you have a way to update dependencies other than https://flaviocopes.com/update-npm-dependencies/

RichardMcCarty1 commented:
You can manually change dependencies in the package.json and remove them from node_modules to force them to revert, but beyond that I believe the only method is npm install packageName@version# i.e React@16.8.0.

Supposedly this line will fix this specific issue, seems to be common, from [here](https://stackoverflow.com/questions/64532965/how-to-fix-this-error-while-installing-radium-packages-in-vs-code-terminal-tha)
```npm install --save radium --legacy-peer-deps```

RichardMcCarty1 commented:
I'm pulling down a clean clone as well to see if I have a similar issue and to see about resolving it as well.

bowring commented:
this works: npm install --save radium --legacy-peer-deps. thanks!

RichardMcCarty1 commented:
Yeah my clone came up clean so I suppose it's good that that line worked. If it ends up being a recurring issue in additional install cycles we can look into a method for resolving it permanently.

bowring commented:
UI looking good!

RichardMcCarty1 commented:
I removed the package-lock from my branch; however, because it exists as a file in the master of Cirdles/CirdlesWebUI it's throwing a conflict.

bowring commented:
Spot average choice did not stick

RichardMcCarty1 commented:
@bowring Spot Average should now stick between refreshes / sessions, changed the routing from Skeleton to manageproject, and fixed the Z-index for the dropdown in manageproject. 

Still working through the rest of the list but that's a temporary update.

bowring commented:
Wrong file is specified in Project File item - it shows previous quid file name

bowring commented:
also, get this when try to serially open files:

Cannot GET /squidink/manageproject/manageproject

RichardMcCarty1 commented:
For the project file name, are you saying that when you open a second project the name of the first project you opened is persisting? And for the serially opening files, I take it you're opening a demo file from the manageproject screen? 

I can't gain reference to the standard react-router format for redirecting users from the constants file where the POST request for demo file is stored so I just manually force an update to the href so I'll add in a check to see if the user is still on manageprojects and if so push a refresh instead.

bowring commented:
it appears above is related to file manager quitting.  However, If I open a project and then try to open demofile, I get the error:
Cannot GET /squidink/manageproject/manageproject

bowring commented:
Nice! - all seems well except file manager menu seems dead now

bowring commented:
Much better!

bowring commented:
maybe we should have a Files / Manage Files menu item to the left of Project that opens the filebrowser

RichardMcCarty1 commented:
I can look into it, there was a point in time where you could open/close the filebrowser from a dropdown option in the Project dropdown, I'll have to add the dropdown to the manageproject page but I don't see it being an issue.

bowring commented:
at least when the default project is loaded on top of another project.

there is another project you can use inside the resources of squidcore to check this

bowring commented:
Still have issue with Project File text from above.

bowring commented:
Project page border looks good; still needs scroll when slightly collapsed

RichardMcCarty1 commented:
Yeah I haven't been able to investigate the project file issue but its next on my docket.

bowring commented:
I like!  The window is well-behaved with resizing, which is also great!

bowring commented:
that fixes the fie loading!

bowring commented:
Looks like we lost the layout and scrolling part of manageproject;  see my pull request to this branch for updates re: about etc.

bowring commented:
Looking good so far!

bowring commented:
Nice!

bowring commented:
something is amiss ... I change analyst name and it save it to the projrct name then the file name is changed to something else?

bowring commented:
looking good

Issue: the two spinners need refinement: 
1) they jump around a little when plus sign is pressed
2) they should be constrained to [0,1] (inclusive) and show only 0.00, 0.05, 0.10, 0.15 ... 1.00

RichardMcCarty1 commented:
The most recent push introduces all the model / model data handling and generation as well as some more linking between the SquidAPI and the components within the Spot Management page.

Next push should finish the Spot Renaming functionality, wrap up the left-side context menu as well as the (C)RM context menus, and handle the information received from the spot audit to generate error/warning text.

bowring commented:
Thanks!

bowring commented:
looking good

bowring commented:
looking good

bowring commented:
View current task not doing anything, and by had, get: Cannot GET /squidink/currenttask

bowring commented:
getting no file selected screen for currenttask after last update

RichardMcCarty1 commented:
It should fix itself if you pull down the new WebServices and build that and then open a file. I introduced the mount variable which ensures its pulled all the requisite information before rendering the page as it is in all our other pages, so if your WebServices isn't updated, it won't be able to pull all the info and the mount will never toggle to true.

bowring commented:
you mean CirdlesWebServices or SquidInkWebServices

On Sun, Nov 21, 2021 at 9:42 AM RichardMcCarty1 ***@***.***>
wrote:

> It should fix itself if you pull down the new WebServices and build that
> and then open a file. I introduced the mount variable which ensures its
> pulled all the requisite information before rendering the page as it is in
> all our other pages, so if your WebServices isn't updated, it won't be able
> to pull all the info and the mount will never toggle to true.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CIRDLES/CirdlesWebUI/pull/100#issuecomment-974829467>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIK2YYMZ6GTJMGSIXIUJVTUNEAM7ANCNFSM44X2I7MQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Jim Bowring
Associate Professor Emeritus, Computer Science, College of Charleston
Principal Investigator, www.CIRDLES.org <http://www.cirdles.org/>,
www.github.com/cirdles
GitHub: https://github.com/bowring

R. Buckminster Fuller (1972):
If humanity is to survive aboard our planet, it must become universally
literate and preoccupied with inherently cooperative Comprehensive
Anticipatory Design Science in which every human is concerned with
accomplishing the comfortably sustainable well-faring of all other humans.

RichardMcCarty1 commented:
SquidinkWebServices, I forgot to actually put my branch up on a PR because we had had the long-running PRs for so long. It's up now though.

bowring commented:
looking good

RichardMcCarty1 commented:
This contains the new route and the skeleton for all of the content that'll be filled when certain tasks are selected. Currently doesn't really do anything besides rendering the default tasks folder and the skeleton for the actual task viewing. The servlet exists to pull the information for each task name so I'll link that up in the coming days and work on rendering the mass labels and ratios as well as all the generic stuff that comes from currentTask.

It seems I can render ellipses and boxes with an SVG tag, so with some styling it should be able to look almost exactly like what is in Squid currently.

bowring commented:
Cool - I had to type tasklibrary into the address bar as menu is disabled.

bowring commented:
three items:
1) list of tasks should be in intuitive-alpha order
2) directive expressions are not updating correctly with change in selected task
3) third button for using task is missing

great work!
